crates/kubelet/src/state.rs

```diff
@@ -28,8 +28,12 @@ pub struct StateHolder<PodState> {
 pub enum Transition<PodState> {
     /// Transition to new state.
     Next(StateHolder<PodState>),
-    /// This is a terminal node of the state graph.
+    /// Stop executing the state machine and report the result of the execution.
     Complete(anyhow::Result<()>),
+    /// Stop executing the state machine and report that the state machine
+    /// failed.
+    // TODO: is this really any different from Complete(Err(...))?
+    Fatal(anyhow::Error),
```

Comment:
I tend to agree that `Complete(Err())` could be used for this.

Reply:
I second

Reply:
Yep, done